src/bridge/cockpitstream.c

```diff
@@ -745,6 +790,19 @@ cockpit_stream_class_init (CockpitStreamClass *klass)
                                          NULL, NULL, NULL,
                                          G_TYPE_NONE, 1, G_TYPE_STRING);
 
+  /**
+   * CockpitStream::pressure:
+   * @throttle: Pressure on or off
+   *
+   * Emitted when the websocket wants to give back-pressure to other feeding
+   * streams. It does this when its output queue is too long and should
+   * slow down.
+   */
+  cockpit_stream_sig_pressure = g_signal_new ("pressure", COCKPIT_TYPE_STREAM, G_SIGNAL_RUN_FIRST,
```

Comment:
I don't like the adhoc way you're defining this same thing in multiple places and then effectively relying on ducktyping when you go to hook stuff up.  Why not use an interface here?

Similarly: you could have a second interface for the other side of it (ie: objects that can receive pressure signals) and use the interface code to avoid some duplication of the weak pointer and signal hookup code.

Reply:
>  Why not use an interface here?

Sure, will do. An interface can reduce code. But almost any use of signals in a GLib program is duck-typing. The WebSocket code has also been logically separate, and someone can combine that separately. So that part will remain duck typed ... but that does not affect this pull request.

> Similarly: you could have a second interface for the other side of it (ie: objects that can receive pressure signals) and use the interface code to avoid some duplication of the weak pointer and signal hookup code.

That's not an interface. That would be a multi-inheritance base class ... or at best a strange helper function that takes a partial struct (ie: a mixin).